hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -226,7 +226,7 @@ instance Arbitrary ChainContext where
   arbitrary = do
     networkId <- Testnet . NetworkMagic <$> arbitrary
     ownVerificationKey <- genVerificationKey
-    otherParties <- arbitrary
+    otherParties <- arbitrary `suchThat` (\a -> not (null a) && length a <= maximumNumberOfParties)
```

Comment:
nice catch!

Reply:
@v0d1ch Why did you need this / got tripped by this? `otherParties` being `[]` or having `10` entries is a perfectly fine `Arbitrary ChainContext`. IIRC we usually don't use the `arbitrary` generator, but `deriveChainContexts` or `pickChainContext` and a generated `HydraContext` in tests.

Reply:
I needed this one when creating valid arbitrary instance for `TxOutWithWitnesses` at some point. The problem was that the line below `ownParty <- elements otherParties` depends on this list being populated and it seems previously we didn't use these arbitrary values anywhere but now we do in the `TxSpec`.